blockchain/chain/src/store/chain_store.rs

```diff
@@ -210,6 +235,51 @@ where
 
         Ok(FullTipset::new(blocks))
     }
+    /// Determines if provided tipset is heavier than existing known heaviest tipset
+    fn is_heaviest(&mut self, ts: &Tipset) -> Result<(), Error> {
+        // TODO determine if expanded tipset is required; see https://github.com/filecoin-project/lotus/blob/testnet/3/chain/store/store.go#L236
+        let new_weight = self.weight(ts)?;
+        let curr_weight = self.weight(&self.heaviest)?;
+
+        if new_weight > curr_weight {
+            // TODO potentially need to deal with re-orgs here
+            info!("New heaviest tipset");
+            self.set_heaviest_tipset(Arc::new(ts.clone()))?;
+        }
+
+        Ok(())
+    }
```

Comment:
Maybe change the name of this to update_heaviest() or something, confusing the actual function of this